src/libraries/System.Private.CoreLib/src/System/Threading/ThreadInt64PersistentCounter.cs

```diff
@@ -9,16 +9,24 @@ namespace System.Threading
 {
     internal sealed class ThreadInt64PersistentCounter
     {
-        private static readonly LowLevelLock s_lock = new LowLevelLock();
+        private readonly LowLevelLock _lock = new LowLevelLock();
 
         [ThreadStatic]
         private static List<ThreadLocalNodeFinalizationHelper>? t_nodeFinalizationHelpers;
 
         private long _overflowCount;
 
-        // we pass comparer explicitly so that in NativeAOT case we would not end up
-        // calling into the type system when lock contention increments its counter
-        private HashSet<ThreadLocalNode> _nodes = new HashSet<ThreadLocalNode>(ObjectEqualityComparer<object>.Default);
+        // dummy node serving as a start of the list
+        private ThreadLocalNode _first;
+
+        public ThreadInt64PersistentCounter()
+        {
+            _first = new ThreadLocalNode(this);
+            // terminator dummy, to simplify inserting/removing
+            ThreadLocalNode last = new ThreadLocalNode(this);
+            _first._next = last;
+            last._prev = _first;
```

Comment:
```suggestion
            _first._next = _first;
            _first._prev = _first;
```
Can we get away with just dummy by setting up as a ring?